sklearn/decomposition/_kernel_pca.py

```diff
@@ -346,6 +346,19 @@ def transform(self, X):
     def inverse_transform(self, X):
         """Transform X back to original space.
 
+        ``inverse_transform`` approximates the inverse transformation using
+        a learned pre-image. The pre-image is learned by kernel ridge
+        regression of the original data on their low-dimensional representation
+        vectors.
+
+        .. note::
```

Comment:
Let's add such note in the `kernel` presentation parameter. We can add something like:

```
.. note::
   Be aware that `kernel="linear"` is equivalent of using a
   :class:`~sklearn.decomposition.PCA`. You recommend using
   :class:`~sklearn.decomposition.PCA` in this case such that `inverse_transform`
   will be an exact reconstruction without data loss.
```

Reply:
Do you mean moving this note to the description of the parameter `kernel`? I assume we do not have to warn users about `kernel="linear"` as long as they do not use `inverse_transform` so I would suggest leaving it as it is.

Reply:
> Do you mean moving this note to the description of the parameter kernel?

It was what I was thinking.

> I assume we do not have to warn users about kernel="linear" as long as they do not use inverse_transform so I would suggest leaving it as it is.

I would have an apriori thinking that a user might look at the parameter documentation but necessarily to the `inverse_transform` when creating the object.

Reply:
Yes, it makes sense, but at the same time advising the users not to use `kernel="linear"` while using `kernel="linear"` as the default parameter looks very weird. I would add such advice to the description of the `kernel` when the default is no longer `"linear"`.

Reply:
Point taken. We can postpone after the default changes.